dash-pipeline/SAI/templates/dashsai.cpp.j2

```diff
@@ -0,0 +1,649 @@
+#include "dashsai.h"
+
+#include <cstdlib>
+
+using namespace dash;
+
+#define DASH_BMV2_CPU_QOS_NUMBER_OF_QUEUES 0
+
+#define DASH_OBJECT_SHFT 48
+#define DASH_MAKE_OID(objtype, objval) (sai_object_id_t)(((sai_object_id_t)objtype<<DASH_OBJECT_SHFT)+(sai_object_id_t)objval)
+
+#define DASH_CHECK_API_INITIALIZED()                                        \
+    if (!m_apiInitialized) {                                                \
+        DASH_LOG_ERROR("%s: api not initialized", __PRETTY_FUNCTION__);     \
+        return SAI_STATUS_FAILURE; }
+
+/**
+ * @def DASH_USE_NOT_SUPPORTED
+ *
+ * Name of environment variable, when set to any value, will return proper
+ * SAI_STATUS_NOT_SUPPORTED on switch/port GET api.
+ *
+ * Currently because of limitation of saithrift, GET on switch/port is
+ * returning SAI_STATUS_SUCCESS on not implemented attribute.
+ *
+ * TODO FIXME needs to be addressed in saithrift.
+ *
+ * This is temporary workaround for syncd, it should be removed when fixed in saithrift.
```

Comment:
Is there a corresponding issue filed against saithrift?

Reply:
not yet